lib/OpenQA/Task/Job/HookScript.pm

```diff
@@ -1,28 +1,40 @@
-# Copyright 2022 SUSE LLC
+# Copyright 2022 SUSE LLC3
 # SPDX-License-Identifier: GPL-2.0-or-later
 
 package OpenQA::Task::Job::HookScript;
 use Mojo::Base 'Mojolicious::Plugin', -signatures;
 use OpenQA::Task::SignalGuard;
+use Data::Dumper;
+$Data::Dumper::Indent = 1;
+$Data::Dumper::Sortkeys = 1;
+
+use constant LINEAR_BACKOFF_BASE_SECONDS => 40;
 
 sub register ($self, $app, $config) {
     $app->minion->add_task(hook_script => \&_hook_script);
 }
 
 sub _hook_script ($job, $hook, $openqa_job_id, $options) {
     my $ensure_task_retry_on_termination_signal_guard = OpenQA::Task::SignalGuard->new($job);
-    my $timeout = $options->{timeout};
-    my $kill_timeout = $options->{kill_timeout};
-    my $delay = $options->{delay};
-    my $retries = $options->{retries};
-    my $skip_rc = $options->{skip_rc};
+    my $timeout = $options->{timeout};    # 5
+    my $kill_timeout = $options->{kill_timeout};    # 30
+
+    #my $delay = $options->{delay};    # 1min
+    my $retries = $options->{retries};    # 1440
+    my $delay = ($options->{delay} == 0) ? $options->{delay} : (LINEAR_BACKOFF_BASE_SECONDS * ($job->retries + 1));
```

Comment:
If the delay is 0 you want to use the delay which is 0?

Reply:
yes. 
`$options->{delay}` seems it comes from `$settings->{_TRIGGER_JOB_DONE_DELAY} // $ENV{OPENQA_JOB_DONE_HOOK_DELAY} // ONE_MINUTE`, so there are cases where the $ENV{OPENQA_JOB_DONE_HOOK_DELAY} is set to 0 to trigger a retry immediately afait from the tests.